spring-integration-core/src/main/java/org/springframework/integration/codec/kryo/AbstractKryoCodec.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.codec.kryo;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.esotericsoftware.kryo.pool.KryoCallback;
+import com.esotericsoftware.kryo.pool.KryoFactory;
+import com.esotericsoftware.kryo.pool.KryoPool;
+
+import org.springframework.integration.codec.Codec;
+import org.springframework.util.Assert;
+
+
+/**
+ * Base class for Codecs using {@link com.esotericsoftware.kryo.Kryo}. Manages pooled Kryo
+ * instances.
+ * @author David Turanski
```

Comment:
`@since 4.2` (or `4.1.7` if we approve) and no reason to have fully qualified class name above, because we `import` it and use in this class.
Minor, of course, and it may be just my caprice which I can fix on merge...